src/main/java/mods/railcraft/data/recipes/providers/BlastFurnaceRecipeProvider.java

```diff
@@ -0,0 +1,92 @@
+package mods.railcraft.data.recipes.providers;
+
+import java.util.function.Consumer;
+import mods.railcraft.Railcraft;
+import mods.railcraft.data.recipes.BlastFurnaceRecipeBuilder;
+import mods.railcraft.world.item.RailcraftItems;
+import mods.railcraft.world.level.block.RailcraftBlocks;
+import net.minecraft.data.DataGenerator;
+import net.minecraft.data.recipes.FinishedRecipe;
+import net.minecraft.data.recipes.RecipeProvider;
+import net.minecraft.resources.ResourceLocation;
+import net.minecraft.world.item.Items;
+import net.minecraft.world.item.crafting.Ingredient;
+import net.minecraft.world.level.ItemLike;
+import net.minecraftforge.common.Tags;
+import net.minecraftforge.registries.ForgeRegistries;
+
+public class BlastFurnaceRecipeProvider extends RecipeProvider {
```

Comment:
Can this stay in `RailcraftRecipeProvider` rather than having multiple `RecipeProvider`s?

Reply:
It is better to divide as they are recipes not from the crafting table. Furthermore, we avoid writing too much in that file which will already become very long

Reply:
Alternatively, make these static utility methods but call them from `RailcraftRecipeProvider`

Reply:
It can be done, but I don't think it's worth it. Better to leave as is as we need the RecipeProvider constructor as it instantiates `recipeProvider` and `advancementProvider`

Reply:
I just don't want multiple data providers registered for recipes (following Minecraft's convention)